godot_project/editor/input_handlers/molecular_structures/add_posing_atom_input_handler.gd

```diff
@@ -82,50 +73,62 @@ func _on_current_structure_context_changed(in_context: StructureContext) -> void
-				var atom_pos: Vector3 = hovered_candidate.atom_position
-				var element_to_create: int = _element_selected
-				const CARBON: int = 6
-				const NITROGEN: int = 7
-				const OXIGEN: int = 8
-				match hovered_candidate.atom_ids.size():
-					1: # Whatever user selected
-						element_to_create = _element_selected
-					2: # Merging 2 candidates, use Oxygen
-						element_to_create = OXIGEN
-					3: # Merging 3 candidates, use Nitrogen
-						element_to_create = NITROGEN
-					4, _: # Merging 4 or more candidates, use Carbon
-						element_to_create = CARBON
-				var params := AtomicStructure.AddAtomParameters.new(element_to_create, atom_pos)
-				var new_bond_order: int = out_context.workspace_context.create_object_parameters.get_new_bond_order()
-				if hovered_candidate.atom_ids.size() > 1:
-					# It's a merge candidate, use bond order 1
-					new_bond_order = 1
-				var _result: Dictionary = _do_create_atom_and_bonds(out_context, params, hovered_candidate.atom_ids, new_bond_order)
-				_ensure_create_mode()
-				_workspace_context.snapshot_moment("Add Atom")
-				return true
-	return false
+	
+	if is_shortcut_pressed:
+		_ensure_create_mode()
+	
+	update_preview_position()
+	rendering.atom_autopose_preview_show()
+
+	if not (in_input_event is InputEventMouseButton and 
+			in_input_event.pressed and
+			in_input_event.button_index == MOUSE_BUTTON_LEFT):
+		return false
+	
+	var preview_atomic_number: int = get_workspace_context().create_object_parameters.get_new_atom_element()
+	var cannot_create_because_hydrogen: bool = not out_context.nano_structure.are_hydrogens_visible() \
+			and preview_atomic_number == PeriodicTable.ATOMIC_NUMBER_HYDROGEN
+	if cannot_create_because_hydrogen:
+		return true
+	
+	var hovered_candidate: AtomCandidate = rendering.atom_autopose_get_hovered_candidate_or_null()
+	if hovered_candidate == null:
+		return false
+	
+	var atom_pos: Vector3 = hovered_candidate.atom_position
+	var element_to_create: int = _element_selected
+	const CARBON: int = 6
+	const NITROGEN: int = 7
+	const OXIGEN: int = 8
+	match hovered_candidate.atom_ids.size():
+		1: # Whatever user selected
+			element_to_create = _element_selected
+		2: # Merging 2 candidates, use Oxygen
+			element_to_create = OXIGEN
+		3: # Merging 3 candidates, use Nitrogen
+			element_to_create = NITROGEN
+		4, _: # Merging 4 or more candidates, use Carbon
+			element_to_create = CARBON
+	var params := AtomicStructure.AddAtomParameters.new(element_to_create, atom_pos)
+	var new_bond_order: int = out_context.workspace_context.create_object_parameters.get_new_bond_order()
+	if hovered_candidate.atom_ids.size() > 1:
+		# It's a merge candidate, use bond order 1
+		new_bond_order = 1
+	var _result: Dictionary = _do_create_atom_and_bonds(out_context, params, hovered_candidate.atom_ids, new_bond_order)
+	_ensure_create_mode()
+	_workspace_context.snapshot_moment("Add Atom")
+	return true
```

Comment:
Actual code didn't change, I just added early returns to lower the indentation levels